sdk/servicebus/azure_messaging_servicebus/src/sender.rs

```diff
@@ -0,0 +1,855 @@
+/// let options = ScheduleMessagesOptions::default();
+/// // Currently no options are available, but this provides
+/// // extensibility for future parameters
+/// ```
+#[derive(SafeDebug, Clone, Default)]
+pub struct ScheduleMessagesOptions {
+    // Currently no options, but preserves API for future extensibility
+}
+
+/// Options for canceling scheduled messages.
+///
+/// This struct contains optional parameters that can be specified when canceling
+/// scheduled messages using [`Sender::cancel_scheduled_message`].
+///
+/// # Examples
+///
+/// ```rust,no_run
+/// use azure_messaging_servicebus::CancelScheduledMessagesOptions;
+///
+/// let options = CancelScheduledMessagesOptions::default();
+/// // Currently no options are available, but this provides
+/// // extensibility for future parameters
+/// ```
+#[derive(SafeDebug, Clone, Default)]
+pub struct CancelScheduledMessagesOptions {
+    // Currently no options, but preserves API for future extensibility
+}
+
+/// A sender for sending messages to a Service Bus queue or topic.
+pub struct Sender {
+    connection: Arc<AmqpConnection>,
+    entity_name: String,
+    _options: ClientOptions,
+}
+
+impl Sender {
+    /// Creates a new sender.
+    pub(crate) async fn new(
+        connection: Arc<AmqpConnection>,
+        entity_name: String,
+        options: ClientOptions,
+    ) -> Result<Self> {
+        debug!("Creating Sender for entity: {}", entity_name);
+
+        trace!("Sender created successfully for entity: {}", entity_name);
+
+        Ok(Self {
+            connection,
+            entity_name,
+            _options: options,
+        })
+    }
+
+    /// Sends a single message to the Service Bus entity.
+    ///
+    /// # Arguments
+    ///
+    /// * `message` - The message to send
+    /// * `options` - Optional parameters for the send operation
+    ///
+    /// # Examples
+    ///
+    /// ```rust,no_run
+    /// use azure_messaging_servicebus::{Message, SendMessageOptions, CreateSenderOptions};
+    /// # use azure_messaging_servicebus::Client;
+    /// # #[tokio::main]
+    /// # async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    /// # let client = Client::from_connection_string("connection_string", Default::default()).await?;
+    /// # let sender = client.create_sender("queue", CreateSenderOptions::default()).await?;
+    ///
+    /// let message = Message::new("Hello, World!".as_bytes());
+    /// let options = SendMessageOptions::default();
+    /// sender.send_message(message, &options).await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn send_message(
+        &self,
+        message: Message,
+        _options: &SendMessageOptions,
+    ) -> Result<()> {
+        debug!("Sending message to entity: {}", self.entity_name);
+
+        // Create AMQP session
+        let session = AmqpSession::new();
+        session.begin(&self.connection, None).await?;
+
+        // Create AMQP sender
+        let amqp_sender = AmqpSender::new();
```

Comment:
I'm not seeing any authorization of the sender on this new `AmqpSender` - does eventhubs allow anonymous access to senders?

Also, establishing a sender is relatively expensive - does it make sense to create the session and sender for every message send?